components/hab/src/cli.rs

```diff
@@ -858,7 +861,12 @@ pub fn sub_sup_run() -> App<'static, 'static> {
         (@arg PEER_WATCH_FILE: --("peer-watch-file") +takes_value conflicts_with[peer]
             "Watch this file for connecting to the ring"
         )
-        (@arg RING: --ring -r +takes_value "Ring key name")
+        (@arg RING: --ring -r env(RING_ENVVAR) conflicts_with("RING_KEY")
+            "The name of the ring used by the Supervisor when running with wire encryption. \
+             (ex: hab sup run --ring myring)")
+        (@arg RING_KEY: --("ring-key") env(RING_KEY_ENVVAR)
+            "The contents of the ring key when running with wire encryption. \
+             (ex: hab sup run --ring-key $(cat /hab/cache/keys/ring-key-file))")
```

Comment:
Have you tested this out? I'm almost certain you'd need quotes around the `$(cat …)` or else the contents would be interpreted by the shell in a way you don't want.

Also, since that example is `bash`-specific, I wonder if @mwrock could supply a Windows-equivalent.

Reply:
Good catch, I missed that. Honestly I'm not really jazzed about the example at all. I think it makes too many assumptions, but I'm not sure what else to show that would be helpful.

Reply:
I think it serves the purpose of making clear that the argument value should be the *contents* of the ring key. Another way to go would be fully explicit:
```
hab sup run --ring-key "SYM-SEC-1
foo-20181113185935

GCrBOW6CCN75LMl0j2V5QqQ6nNzWm6and9hkKBSUFPI="
```

This would also be a good opportunity to put in a validator.

Reply:
I'm gonna make this a hidden option. @christophermaier has some concerns over it's use that I agree with, but I would still like to keep the functionality provided by clap. I think hiding it alleviates both of those issues.

Reply:
It's hidden, but I think it's still useful to add your example if only to help us.

Reply:
I'm at a loss for why we have this option at all. The `--ring` option takes just the name, and we know where to find that file, but `--ring-key` seems like it could just as easily take a path to a file that contains the contents rather than the contents themselves. I imagine that would allay @christophermaier's concerns. And if for whatever reason someone was generating that data on the fly, they could still use something like [process substitution](https://www.gnu.org/software/bash/manual/html_node/Process-Substitution.html):
```bash
$ hab sup run --ring-key <(command-that-generates-key-contents)
```

Reply:
I agree, I was modelling after the existing behavior. It would make more sense to take a path instead of the file contents. I can change the behavior, but that would be a breaking change. While I'm at it the option/variable names seem a bit confusing too. Maybe we should change those to something more descriptive?

Maybe something like: 
`HAB_RING` / `--ring`   to   `HAB_RING_KEY` / `--ring-key`
`HAB_RING_KEY` / `--ring-key`   to   `HAB_RING_KEY_PATH` / `--ring-key-path`